### PR TITLE
feature/SearchByAnimalFilter

### PR DESCRIPTION
**Pull request by SRattacasa**
- Created some psuedo-code options for filtering based on dog or cat
- Evaluating use of switch statement for simplified logic/code

### CODE REVIEW

**Comment by SRattacasa** [comment]
- Built the switch case logic but commented it out for now
- It validates if the dogFilter or catFilter are true/false, and if true, or if both are true, it will add the groomer to the filtered array
- If not, it shouldn't 
- That array is then later sorted for distance based upon the search zipcode/location data

**Comment by SRattacasa** [comment]
- useEffect hook to load groomers was failing at times
- Updated the getGroomerData() function in the api/index.js file to be async
- Haven't had the issue since

**Comment by SRattacasa** [comment]
- Updated and validated the switch case logic
- Applies the proper filter to the results
- Conditionally renders a text statement about the animals groomed if the dogs or cats filter is true

**Comment by SRattacasa** [comment]
- Added states for the checkboxes to update when checked
- Using the states to reset the form by adding setState(false) to the reset button onChange handler
- Removed a lot of old code that was not in use after testing/verifying 
- Verified conditional render logic is working for animals serviced

File: src/components/search/SearchForm.js [hunk 1]
```diff
@@ -64,29 +64,36 @@ function distance(lat1, lon1, lat2, lon2, unit) {
...
-  const handleName = e => setName(e.target.value);
-  const handleZipCode = e => setZipcode(e.target.value);
-  const onSubmit = e => {
-    e.preventDefault();
-    console.log(name, zipcode);
+  // OnChange for the cat checkbox
+  const changeCatFilter = () => {
+    catFilter = !catFilter;
+    setCatState(!catState);
   };
```
Comment (taylorpalmer) [inline src/components/search/SearchForm.js:79]: Your dog/cat logic looks great! But do we not need these lines 81-88?

**Review by taylorpalmer — COMMENTED** [review]
Looks good. Because I'm not familiar, I just want confirmation that we won't miss the err handling etcetera that was removed before I approve.

File: src/components/search/SearchForm.js [hunk 2]
```diff
@@ -64,29 +64,36 @@ function distance(lat1, lon1, lat2, lon2, unit) {
...
 }
-const groomersPerPage = 3;
+
+// Filter variables for animal checkbox
+let dogFilter = false;
+let catFilter = false;
+
+const groomersPerPage = 12;
 const SearchForm = props => {
   const [name, setName] = useState('');
```
Comment (coryortega) [inline src/components/search/SearchForm.js]: Great work, remove unused variables and we're all good!

**Review by coryortega — CHANGES_REQUESTED** [review]
Remove unused variables

File: src/components/search/SearchForm.js [hunk 3]
```diff
@@ -101,23 +92,46 @@ const SearchForm = props => {
...
+          if (groomer.cats === true) {
+            filtered.push(groomer);
+          }
+          break;
+
+        default:
+          filtered.push(groomer);
+          break;
+      }
+
```
Comment (coryortega) [inline src/components/search/SearchForm.js:125]: Nice use of switch cases!

**Review by coryortega — APPROVED** [review]
Nice work! Clean code, nice use of switch cases

**Review by AustinKelsay — APPROVED** [review]

**Pull request merged**